### ISSUE TITLE
add custom loss 

### ISSUE THREAD

**Issue by MisayaZ** [tensorflow/tensorflow#6301]
I have noticed that tensorflow can add custom op, however, I want to add an custom loss function. To my
point of view, an custom op and an custom loss are different. Dose tensorflow support adding an custom loss?

**Comment by bc-lee** [comment]
I think you could do it by subclassing `tf.train.Optimizer` class. ref: https://github.com/tensorflow/tensorflow/blob/72a3caa1cc223fe10ac5287083ef20a911f87400/tensorflow/python/training/optimizer.py#L364

**Comment by MisayaZ** [comment]
@bc-lee Thank you for your suggestion. Do you mean modify the source code of tensorflow?

**Comment by andydavis1** [comment]
This is a question better suited for StackOverflow, which we also monitor. Please ask it there and tag it with the `tensorflow` tag.
